src/Core/Managed/Shared/Extensibility/MetricManager.cs

```diff
@@ -119,12 +119,6 @@ public void Flush()
         /// </summary>
         public void Dispose()
         {
-            if (this.snapshotTimer != null)
-            {
-                this.snapshotTimer.Dispose();
-                this.snapshotTimer = null;
-            }
-
             this.Flush();
```

Comment:
It appears to me that in case the only job of `MetricManager.Dispose()` class is to call `Flush()`, we can make `MetricManager` itself non-disposable?